src/DynamicNLPModels.jl

```diff
@@ -2207,14 +2197,72 @@ end
 """
     LinearOperators.reset!(Jac::LQJacobianOperator{T, V, M})
 
-Resets the values of attributes `JB1`, `JB2`, and `JB3` to zero
+Resets the values of attributes `SJ1`, `SJ2`, and `SJ3` to zero
 """
 function LinearOperators.reset!(
     Jac::LQJacobianOperator{T, V, M}
 ) where {T, V <: AbstractVector{T}, M <: AbstractMatrix{T}}
-    fill!(Jac.JB1, T(0))
-    fill!(Jac.JB2, T(0))
-    fill!(Jac.JB3, T(0))
+    fill!(Jac.SJ1, T(0))
+    fill!(Jac.SJ2, T(0))
+    fill!(Jac.SJ3, T(0))
+end
+
+"""
+    add_jtsj!(H::M, Jac::LQJacobianOperator{T, V, M}, Σ::V)
+
+Generates `Jac' Σ Jac` and adds it to the matrix `H`.
+"""
+function add_jtsj!(
```

Comment:
I would suggest we update the signature of `add_jtsj!` to: 
```julia 
add_jtsj!(H::M, Jac::LQJacobianOperator{T, V, M}, Σ::V, alpha::Number, beta::Number)
```
That way, we can choose to update `H` inplace `beta=1` or to overwrite `H` (`beta=0`)